stage1/init/common/pod.go

```diff
@@ -256,28 +259,61 @@ func appToSystemd(p *stage1commontypes.Pod, ra *schema.RuntimeApp, interactive b
 		return errwrap.Wrap(errors.New("unable to write environment file"), err)
 	}
 
-	// This is a partial implementation for app.User and app.Group:
-	// For now, only numeric ids (and the string "root") are supported.
-	var uid, gid int
+	var _uid, gid int
 	var err error
-	if app.User == "root" {
-		uid = 0
+
+	uidRange := uid.NewBlankUidRange()
+	if err := uidRange.Deserialize([]byte(privateUsers)); err != nil {
+		return errwrap.Wrap(fmt.Errorf("unable to deserialize uid range"), err)
```

Comment:
What is the rule between `errors.New` and `fmt.Errorf`? Does it depend on whether the string needs formatting (with %s or %d)?